lib/plug/builder.ex

```diff
@@ -212,11 +212,21 @@ defmodule Plug.Builder do
       :function -> "expected #{inspect plug}/2 to return a Plug.Conn"
     end
 
+    halt_message = case plug_type do
+      :module   -> "plug pipeline halted in #{inspect plug}.call/2"
+      :function -> "plug pipeline halted in #{inspect plug}/2"
+    end
+
     quote do
       case unquote(compile_guards(call, guards)) do
-        %Plug.Conn{halted: true} = conn -> conn
-        %Plug.Conn{} = conn             -> unquote(acc)
-        _                               -> raise unquote(error_message)
+        %Plug.Conn{halted: true} = conn ->
+          require Logger
+          Logger.info unquote(halt_message)
```

Comment:
@whatyouhide @chrismccord should this behaviour be opt-in? Otherwise plug would log every time an static asset is served. Maybe it could be an option called `log_on_halt: :debug`, where the value is the level which defaults to nil?